src/ripple/app/tx/impl/SetAccount.cpp

```diff
@@ -176,6 +176,21 @@ SetAccount::doApply ()
     bool bSetDisallowXRP   = (uTxFlags & tfDisallowXRP) || (uSetFlag == asfDisallowXRP);
     bool bClearDisallowXRP = (uTxFlags & tfAllowXRP) || (uClearFlag == asfDisallowXRP);
 
+    bool sigWithMaster = false;
+
+    {
+        auto const blob = ctx_.tx.getSigningPubKey();
+
+        if (!blob.empty ())
+        {
+            auto const signingPubKey =
+                RippleAddress::createAccountPublic(blob);
+
+            if (calcAccountID(signingPubKey) == account_)
+                sigWithMaster = true;
+        }
```

Comment:
This check looks right to me.  The only thing I wondered about is whether we need to catch any exceptions.  I did a quick visual audit and, as far as I can see, there are no `throw`s coming from the called functions.  So I think you're good to go.